proc_parent.go

```diff
@@ -466,3 +471,26 @@ func extension() string {
 
 	return ""
 }
+
+func (mp *parent) overwriteBinary(tmpBinPath string) error {
+	return mp.withFileLock(func() error {
+		return overwrite(mp.binPath, tmpBinPath)
+	})
+}
+
+func (mp *parent) withFileLock(fn func() error) error {
+	// Use the binary itself as a lock file to prevent other instances from
+	// trying to fetch updates at the same time.
+	touchFile, err := touchfile.NewTouchFile(mp.binPath)
```

Comment:
I tried looking through the `touchfile` docs / implementation but couldn't find the answer - does unlocking the touchfile remove it?

I ask because my mental model is:
* lock file exists - lock is acquired by another process
* lock file does not exist - no other process has the lock

Reply:
This is using the `binPath` - the binary itself - as the file being locked. And no, it does **not** remove it when it is unlocked.

Reply:
This got me thinking, and I suspect that the tactic of locking the binary itself is not going to work in practice, because the file being locked is going to disappear on a successful update. Instead, I'm going to modify this to use a file with a well-known path that persists between updates.